### DISCUSSION TITLE
Submit run programmatically (without dagit graphQL)

### DISCUSSION THREAD

**Discussion by programmervarunshridhar** [dagster-io/dagster#7791] [Q&A]
Hello all,

Anyway I can submit dagster run programmatically without using dagit graphQL?

`   
    return dagster_instance.submit_run(
        pipeline_run.run_id,
        workspace=graphene_info.context,
 )`

Basically I want to use above code to submit a new run but unable to figure out how to get the pipeline_run and workspace.

Appreciate your time and help.

**Comment by yuhan** [comment]
Have you tried the [CLI](https://docs.dagster.io/_apidocs/cli#dagster-job)?
`dagster job launch` is an alternative way to submit run without Dagit or GraphQL.

**Reply by programmervarunshridhar** [reply]
Hello @yuhan 

Thank you for the suggestion.

I am looking for a solution where I can submit run programmatically (via code). This code will reside in a rest API action method.

**Reply by programmervarunshridhar** [reply]
Hello,

I was going through the dagster-graphql code for launchRun mutation.

In the file "launch_execution.py" (https://github.com/dagster-io/dagster/blob/master/python_modules/dagster-graphql/dagster_graphql/implementation/execution/launch_execution.py) there is an implementation for do_launch.

this method accesses some objects from graphene_info which I believe comes from graphene implementation. Any way I can manually create these objects and use them in the `dagster_instance.submit_run` implementation?
